main.go

```diff
@@ -20,7 +20,55 @@ type TraefikOidcAuth struct {
 	Jwks              *JwksHandler
 }
 
+// Make sure we fetch oidc discovery document during first request - avoid race condition
+// Perform lock when changing document - we are in concurrent environment
+func EnsureOidcDiscovery(toa *TraefikOidcAuth) (*TraefikOidcAuth, error) {
+	var config = toa.Config
+	var parsedURL = toa.ProviderURL
+	if toa.DiscoveryDocument == nil {
+		var jwks = &JwksHandler{}
```

Comment:
I think we shouldn't mix the JWKS lock with the discovery document lock.
I'am not a Go-expert, but if we create a new JwksHandler here, then i think we will have a new lock on every call, which makes it useless.
Maybe we create another mutex on the "toa" (TraefikOidcAuth) for getting the discovery document.

Reply:
And: I think after we've acquired the lock, we should check for `if toa.DiscoveryDocument == nil` again.

Reply:
Fixed both of them.
I am creating a JwksHandler since it's needed .